cmd/lakectl/cmd/plugin_list.go

```diff
@@ -0,0 +1,143 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const pluginListTemplate = `The following lakectl-compatible plugins are available:
+
+{{- range .Plugins }}
+  {{- if or .Verbose .Warning }}
+  {{ .Path }}
+  {{- else }}
+  {{ .Name }}
+  {{- end }}
+  {{- if .Warning }}
+    - {{ "warning:" | yellow }} {{ .Warning }}
+  {{- end }}
+{{- end }}
+`
+
+type pluginInfo struct {
+	Name    string
+	Path    string
+	Warning string
+	Verbose bool
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available lakectl plugins",
+	Long:  `Scans the PATH for executables named "lakectl-*" and lists the detected plugins.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		envPathDirs := filepath.SplitList(os.Getenv("PATH"))
+
+		// Filter out duplicates from envPathDirs
```

Comment:
This is not needed, you should anyway take the first occurrence of each executable in PATG.

Reply:
We don't - I don't want to read the directory again